### PR TITLE
CMCL-0000: InspectorUtility API cleanup, mostly cosmetic

### PR DESCRIPTION
**Pull request by glabute**
### Purpose of this PR

Renames, file reordering.  Also removed a small layout hack.

### CODE REVIEW

File: com.unity.cinemachine/Editor/Utility/InspectorUtility.cs [hunk 1]
```diff
@@ -484,68 +501,84 @@ public static void AddSpace(this VisualElement ux)
...
+            {
+                label.AddToClassList("unity-base-field__label--with-dragger");
+                label.OnInitialGeometry(() =>
+                {
+                    if (p.propertyType == SerializedPropertyType.Float)
+                        new FieldMouseDragger<float>(field.Q<FloatField>()).SetDragZone(label);
+                    else if (p.propertyType == SerializedPropertyType.Integer)
+                        new FieldMouseDragger<int>(field.Q<IntegerField>()).SetDragZone(label);
+                });
+            }
```
Comment (gaborkb) [inline com.unity.cinemachine/Editor/Utility/InspectorUtility.cs:519]: These may be a bit cleaner:
```
if (p.propertyType == SerializedPropertyType.Float)
{
    label.AddToClassList("unity-base-field__label--with-dragger");
    label.OnInitialGeometry(() => new FieldMouseDragger<float>(field.Q<FloatField>()).SetDragZone(label));
}
else if (p.propertyType == SerializedPropertyType.Integer)
{
    label.AddToClassList("unity-base-field__label--with-dragger");
    label.OnInitialGeometry(() => new FieldMouseDragger<int>(field.Q<IntegerField>()).SetDragZone(label));
}
```
or
```
switch (p.propertyType)
{
    case SerializedPropertyType.Float:
        label.AddToClassList("unity-base-field__label--with-dragger");
        label.OnInitialGeometry(() => new FieldMouseDragger<float>(field.Q<FloatField>()).SetDragZone(label));
        break;
    case SerializedPropertyType.Integer:
        label.AddToClassList("unity-base-field__label--with-dragger");
        label.OnInitialGeometry(() => new FieldMouseDragger<int>(field.Q<IntegerField>()).SetDragZone(label));
        break;
}
```

**Review by gaborkb — CHANGES_REQUESTED** [review]

File: com.unity.cinemachine/Editor/Utility/InspectorUtility.cs [hunk 2]
```diff
@@ -635,37 +671,25 @@ public CompactPropertyField(SerializedProperty property, string label, float min
...
+                { style = { flexGrow = 1, flexBasis = SingleLineHeight }});
+            AddPropertyDragger(row.Label, property, propertyField);
 
+            // Kill any left margin that gets inserted into the property field
+            field.OnInitialGeometry(() => 
+            {
+                var children = field.Children().GetEnumerator();
+                if (children.MoveNext())
+                    children.Current.style.marginLeft = 0;
+            });
```
Comment (gaborkb) [inline com.unity.cinemachine/Editor/Utility/InspectorUtility.cs:688]: We need to dispose the enumerator, e.g.:
`using var children = field.Children().GetEnumerator();`

**Review by gaborkb — CHANGES_REQUESTED** [review]
This one got lost somehow:

File: com.unity.cinemachine/Editor/Utility/InspectorUtility.cs (hunk 1 above)
Comment (glabute) [reply com.unity.cinemachine/Editor/Utility/InspectorUtility.cs:519]: I don't see how that's cleaner

File: com.unity.cinemachine/Editor/Utility/InspectorUtility.cs (hunk 2 above)
Comment (glabute) [reply com.unity.cinemachine/Editor/Utility/InspectorUtility.cs:688]: good catch.  I didn't know that.  Are there other instances of undisposed GetEnumerator in our code?

Comment (gaborkb) [reply com.unity.cinemachine/Editor/Utility/InspectorUtility.cs]: Will keep an eye out / search for them.

File: com.unity.cinemachine/Editor/Utility/InspectorUtility.cs [hunk 3]
```diff
@@ -635,37 +671,26 @@ public CompactPropertyField(SerializedProperty property, string label, float min
...
+            AddPropertyDragger(row.Label, property, propertyField);
 
+            // Kill any left margin that gets inserted into the property field
+            field.OnInitialGeometry(() => 
+            {
+                var children = field.Children().GetEnumerator();
+                if (children.MoveNext())
+                    children.Current.style.marginLeft = 0;
+                children.Dispose();
+            });
```
Comment (gaborkb) [inline com.unity.cinemachine/Editor/Utility/InspectorUtility.cs:689]: Just a note. :)

There are 3 ways to dispose.
1. Dispose manually
```
field.OnInitialGeometry(() => 
{
    var children = field.Children().GetEnumerator();
    if (children.MoveNext())
        children.Current.style.marginLeft = 0;
    children.Dispose(); // call directly
});
```

2. Use variable scope
```
field.OnInitialGeometry(() => 
{
    using var children = field.Children().GetEnumerator(); // this automatically disposes children when children goes out of scope
    if (children.MoveNext())
        children.Current.style.marginLeft = 0;
});
```

3. Define scope
```
field.OnInitialGeometry(() => 
{
    using (var children = field.Children().GetEnumerator()) {
        if (children.MoveNext())
            children.Current.style.marginLeft = 0;
    }
});
```

**Review by gaborkb — APPROVED** [review]

File: com.unity.cinemachine/Editor/Utility/InspectorUtility.cs (hunk 3 above)
Comment (glabute) [reply com.unity.cinemachine/Editor/Utility/InspectorUtility.cs:689]: TIL :)

**Pull request merged**